src/_pytest/terminal.py

```diff
@@ -1086,21 +1093,28 @@ def print_teardown_sections(self, rep: TestReport) -> None:
                 self._tw.line(content)
 
     def summary_failures(self) -> None:
-        self.summary_failures_combined("failed", "FAILURES")
+        style = self.config.option.tbstyle
+        self.summary_failures_combined("failed", "FAILURES", style=style)
 
     def summary_xfailures(self) -> None:
-        self.summary_failures_combined("xfailed", "XFAILURES", "x")
+        show_tb = self.config.option.xfail_tb
+        style = self.config.option.tbstyle if show_tb else "no"
+        self.summary_failures_combined("xfailed", "XFAILURES", style=style)
 
     def summary_failures_combined(
-        self, which_reports: str, sep_title: str, needed_opt: Optional[str] = None
+        self,
+        which_reports: str,
+        sep_title: str,
+        needed_opt: Optional[str] = None,
+        style: Optional[str] = None,
```

Comment:
I think it makes sense to always require `style`, as it is only called from two places and those always pass `style`.

```suggestion
        sep_title: str,
        *,
        style: str,
        needed_opt: Optional[str] = None,        
```